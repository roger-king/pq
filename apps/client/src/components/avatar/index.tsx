import { AvatarIcon } from './avatar';

export default AvatarIcon;
