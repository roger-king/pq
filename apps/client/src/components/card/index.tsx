export * from './question';
export * from './game';
