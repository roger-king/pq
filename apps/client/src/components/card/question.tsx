import React, { PropsWithChildren } from 'react';
import { Box } from 'grommet';

export interface QuestionCardProps {
  elevation?: boolean;
}

export const QuestionCard: React.FC<PropsWithChildren<QuestionCardProps>> = ({
  children,
  elevation,
}: PropsWithChildren<QuestionCardProps>) => {
  return (
    <Box elevation={elevation ? 'medium' : 'none'} pad="medium" border="all" height={{ min: '278px', max: '400px' }}>
      {children}
    </Box>
  );
};
