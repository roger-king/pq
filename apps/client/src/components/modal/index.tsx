import React, { PropsWithChildren } from 'react';
import { Box, Heading, Layer } from 'grommet';

export interface BaseModalProps {
  title: string;
  onClose(): void;
}

const Modal: React.FC<PropsWithChildren<BaseModalProps>> = (props: PropsWithChildren<BaseModalProps>) => {
  const { children, title } = props;

  return (
    <Layer modal>
      <Box fill align="center" justify="center" pad="medium" background="brand">
        <Box direction="row" align="center" justify="between" fill>
          <Heading level="3">{title}</Heading>
        </Box>
        {children}
      </Box>
    </Layer>
  );
};

export default Modal;
