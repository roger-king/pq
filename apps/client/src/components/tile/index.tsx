import { Tile } from './tile.component';

export default Tile;
