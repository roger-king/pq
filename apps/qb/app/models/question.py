from typing import Optional, List
from sqlalchemy import Column, String, Integer, DateTime, event, ForeignKey, Boolean
from sqlalchemy.ext.mutable import MutableDict
from sqlalchemy.dialects.postgresql import JSONB
from sqlalchemy.sql import func
from sqlalchemy.orm import relationship
from sqlalchemy.dialects.postgresql import UUID
from uuid import UUID as UUIDType, uuid4
from . import Base
from pydantic import BaseModel, Json
from datetime import datetime


class Question(Base):
    __tablename__ = "questions"
    id = Column(Integer, primary_key=True, index=True)
    game_id = Column(Integer, ForeignKey("games.id"))
    q = Column(String, nullable=False, unique=True)
    answer = Column(String, nullable=False)
    options = Column(JSONB, nullable=False, default=[])
    created_by = Column(String, nullable=False)
    created_at = Column(DateTime(timezone=False), server_default=func.now())
    updated_at = Column(DateTime(timezone=False), onupdate=func.now())
    deleted_at = Column(DateTime(timezone=False))

    def as_dict(self):
        return {
            "id": self.id,
            "game_id": str(self.game_id),
            "options": self.options,
            "q": self.q,
            "answer": self.answer,
            "created_by": self.created_by,
        }


class QuestionBase(BaseModel):
    game_id: int
    q: str
    answer: str
    options: List[dict]
    created_by: str
    created_at: datetime
    updated_at: Optional[datetime]
    deleted_at: Optional[datetime]


class QuestionSchema(QuestionBase):
    id: int

    class Config:
        orm_mode = True


class CreateQuestionInput(BaseModel):
    game_id: Optional[int]
    q: str
    answer: str
    created_by: str
    options: List[dict]


class UpdateQuestionOptionsInput(BaseModel):
    options: List[dict]
