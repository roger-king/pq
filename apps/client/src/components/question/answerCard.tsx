import React from 'react';
import { Heading, RadioButtonGroup } from 'grommet';

import { QuestionCard } from './../card';
import { Question } from '../../@types';

interface AnswerCardProps extends Partial<Question> {}

export const AnswerCard: React.FC<AnswerCardProps> = ({ q, options }: AnswerCardProps) => {
  return (
    <QuestionCard>
      <Heading level="2">{q}</Heading>
      <RadioButtonGroup id="example2-id" name="question-option" options={options!.map((o) => `${o.key}. ${o.title}`)} />
    </QuestionCard>
  );
};
