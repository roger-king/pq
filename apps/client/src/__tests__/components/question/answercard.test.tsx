import React from 'react';
import renderer from 'react-test-renderer';

import { AnswerCard } from '../../../components/question';

describe('<AnswerCard />', () => {
  test('should display AnswerCard none selected', async () => {
    const tree = renderer
      .create(
        <AnswerCard
          q="What day is it?"
          options={[
            { key: 'A', title: 'Monday' },
            { key: 'B', title: 'Tuesday' },
          ]}
          selectedAnswer={null}
          setSelectedAnswer={() => console.log('Setting')}
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('should display AnswerCard with selected Answer', async () => {
    const tree = renderer
      .create(
        <AnswerCard
          q="What day is it?"
          options={[
            { key: 'A', title: 'Monday' },
            { key: 'B', title: 'Tuesday' },
          ]}
          selectedAnswer="Monday"
          setSelectedAnswer={() => console.log('Setting')}
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
