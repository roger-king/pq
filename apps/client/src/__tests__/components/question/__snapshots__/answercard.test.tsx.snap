// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AnswerCard /> should display AnswerCard none selected 1`] = `
<div
  className="StyledBox-sc-13pk1d4-0 bxevCE"
>
  <div
    className="StyledBox-sc-13pk1d4-0 jlHNLW"
  >
    <h4
      className="StyledHeading-sc-1rdh4aw-0 bpyjiZ"
    >
      What day is it?
    </h4>
    <div
      className="StyledBox-sc-13pk1d4-0 jvjhNK"
      id="example2-id"
    >
      <label
        className="StyledRadioButton__StyledRadioButtonContainer-g1f6ld-0 hnWETW"
        htmlFor="example2-id-A. Monday"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        <div
          className="StyledBox-sc-13pk1d4-0 bHWzdD StyledRadioButton-g1f6ld-5 imYubY"
        >
          <input
            checked={false}
            className="StyledRadioButton__StyledRadioButtonInput-g1f6ld-1 hIKMbD"
            id="example2-id-A. Monday"
            name="question-option"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            type="radio"
            value="A. Monday"
          />
          <div
            className="StyledBox-sc-13pk1d4-0 cMdTKN StyledRadioButton__StyledRadioButtonBox-g1f6ld-4 kGVpuu"
          />
        </div>
        <span
          className="StyledRadioButton__StyledRadioButtonLabel-g1f6ld-2"
        >
          A. Monday
        </span>
      </label>
      <div
        className="StyledBox__StyledBoxGap-sc-13pk1d4-1 ggstca"
      />
      <label
        className="StyledRadioButton__StyledRadioButtonContainer-g1f6ld-0 hnWETW"
        htmlFor="example2-id-B. Tuesday"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        <div
          className="StyledBox-sc-13pk1d4-0 bHWzdD StyledRadioButton-g1f6ld-5 imYubY"
        >
          <input
            checked={false}
            className="StyledRadioButton__StyledRadioButtonInput-g1f6ld-1 hIKMbD"
            id="example2-id-B. Tuesday"
            name="question-option"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            type="radio"
            value="B. Tuesday"
          />
          <div
            className="StyledBox-sc-13pk1d4-0 cMdTKN StyledRadioButton__StyledRadioButtonBox-g1f6ld-4 kGVpuu"
          />
        </div>
        <span
          className="StyledRadioButton__StyledRadioButtonLabel-g1f6ld-2"
        >
          B. Tuesday
        </span>
      </label>
    </div>
  </div>
</div>
`;

exports[`<AnswerCard /> should display AnswerCard with selected Answer 1`] = `
<div
  className="StyledBox-sc-13pk1d4-0 bxevCE"
>
  <div
    className="StyledBox-sc-13pk1d4-0 jlHNLW"
  >
    <h4
      className="StyledHeading-sc-1rdh4aw-0 bpyjiZ"
    >
      What day is it?
    </h4>
    <div
      className="StyledBox-sc-13pk1d4-0 jvjhNK"
      id="example2-id"
    >
      <label
        className="StyledRadioButton__StyledRadioButtonContainer-g1f6ld-0 hnWETW"
        htmlFor="example2-id-A. Monday"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        <div
          className="StyledBox-sc-13pk1d4-0 bHWzdD StyledRadioButton-g1f6ld-5 imYubY"
        >
          <input
            checked={false}
            className="StyledRadioButton__StyledRadioButtonInput-g1f6ld-1 hIKMbD"
            id="example2-id-A. Monday"
            name="question-option"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            type="radio"
            value="A. Monday"
          />
          <div
            className="StyledBox-sc-13pk1d4-0 cMdTKN StyledRadioButton__StyledRadioButtonBox-g1f6ld-4 kGVpuu"
          />
        </div>
        <span
          className="StyledRadioButton__StyledRadioButtonLabel-g1f6ld-2"
        >
          A. Monday
        </span>
      </label>
      <div
        className="StyledBox__StyledBoxGap-sc-13pk1d4-1 ggstca"
      />
      <label
        className="StyledRadioButton__StyledRadioButtonContainer-g1f6ld-0 hnWETW"
        htmlFor="example2-id-B. Tuesday"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        <div
          className="StyledBox-sc-13pk1d4-0 bHWzdD StyledRadioButton-g1f6ld-5 imYubY"
        >
          <input
            checked={false}
            className="StyledRadioButton__StyledRadioButtonInput-g1f6ld-1 hIKMbD"
            id="example2-id-B. Tuesday"
            name="question-option"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            type="radio"
            value="B. Tuesday"
          />
          <div
            className="StyledBox-sc-13pk1d4-0 cMdTKN StyledRadioButton__StyledRadioButtonBox-g1f6ld-4 kGVpuu"
          />
        </div>
        <span
          className="StyledRadioButton__StyledRadioButtonLabel-g1f6ld-2"
        >
          B. Tuesday
        </span>
      </label>
    </div>
  </div>
</div>
`;
